test/e2e/create_test.go

```diff
@@ -160,30 +160,77 @@ func (tc *testContext) testMachineConfiguration(t *testing.T) {
 	}
 	_, err := tc.createWindowsMachineSet(gc.numberOfMachineNodes, false)
 	require.NoError(t, err, "failed to create Windows MachineSet")
-	// We need to cover the case where a user changes the private key secret before the WMCO has a chance to
-	// configure the Machine. In order to simulate that case we need to wait for the MachineSet to be fully
-	// provisioned and then change the key. The correct amount of nodes being configured is proof that the
-	// mismatched Machine created with the mismatched key was deleted and replaced.
-	// Depending on timing and configuration flakes this will either cause all Machines, or all Machines after
-	// the first configured Machines to hit this scenario. This is a platform agonistic test so we run it only on
-	// Azure.
+
+	t.Run("Machine configuration while private key change", tc.testMachineConfigurationWhilePrivateKeyChange)
+
 	machines, err := tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Provisioned", false)
 	require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
-	if tc.CloudProvider.GetType() == config.AzurePlatformType {
-		// Replace the known private key with a randomly generated one.
-		err = tc.createPrivateKeySecret(false)
-		require.NoError(t, err, "error replacing private key secret")
-	}
 	err = tc.waitForConfiguredWindowsNodes(gc.numberOfMachineNodes, false, false)
 	assert.NoError(t, err, "Windows node creation failed")
-	if tc.CloudProvider.GetType() == config.AzurePlatformType {
-		// Update the machines list
-		machines, err = tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Running", false)
-		require.NoError(t, err, "error waiting for Windows Machines to be running")
-	}
 	tc.machineLogCollection(machines.Items)
 }
 
+// testMachineConfigurationWhilePrivateKeyChange tests that machines are deleted after the private key change.
+// We need to cover the case where a user changes the private key secret before the WMCO has a chance to
+// configure the Machine. In order to simulate that case we need to wait for the MachineSet to be fully
+// provisioned and then change the key. The correct amount of nodes being configured is proof that the
+// mismatched Machine created with the mismatched key was deleted and replaced.
+// Depending on timing and configuration flakes this will either cause all Machines, or all Machines after
+// the first configured Machines to hit this scenario. This could be considered a platform-agnostic test, except for
+// vSphere where the private key is baked in the VM template, so we run it only on Azure.
```

Comment:
This doc is a bit overwhelming :smile:
Moving it to the function level, we don't need to go into too much detail, as it might actually make it less clear to the caller what is actually happening.

I suggest trimming it down to something like:
```
// testMachineConfigurationWhilePrivateKeyChange tests that machines which have not yet been configured by WMCO are deleted after the private key is changed, but before WMCO is able to configure them, resulting in WMCO getting an SSH authentication error
```

Reply:
Agree, addressed.